### PR TITLE
feature test sign_in, sign_up and dropdown select project

### PR DESCRIPTION
**Pull request by SetterVet**

### CODE REVIEW

File: spec/feature/feature_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+describe 'Feature', type: :feature do
+  before(:each) do
+    visit ''
```
Comment (VasylYaremkiv) [inline spec/feature/feature_spec.rb]: Do we need this?

File: spec/feature/feature_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  end
+  let(:user) { FactoryBot.create(:user) }
+  context 'open table of users by' do
+    it 'sign in', js: true do
+      expect(page).to have_current_path '/users/sign_in'
```
Comment (VasylYaremkiv) [inline spec/feature/feature_spec.rb]: Feature tests should not check path. They should check view content (https://m.patrikonrails.com/how-i-test-my-rails-applications-cf150e347a6b System tests).
Controller tests should check path.

Comment (VasylYaremkiv) [reply spec/feature/feature_spec.rb]: Use route path like `user_sign_in_path` instead hard code.

File: spec/feature/feature_spec.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      fill_in 'Password', with: 'secret'
+      fill_in 'Password confirmation', with: 'secret'
+      select project.name, from: 'Project'
+      click_button 'Sign up'
+      expect(page).to have_current_path '/'
+      expect(page).to have_content 'List of users'
+    end
+  end
+  # context 'user vue component' do
+  #
```
Comment (VasylYaremkiv) [inline spec/feature/feature_spec.rb]: Remove not used code.

File: spec/feature/sign_in_spec.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+describe 'Sign in', type: :feature do
+  before(:each) do
+    visit root_path
```
Comment (VasylYaremkiv) [inline spec/feature/sign_in_spec.rb]: `visit new_user_session_path`

File: spec/feature/sign_in_spec.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  end
+  let(:user) { FactoryBot.create(:user) }
+  it 'correct', js: true do
+    expect(page).to have_current_path new_user_session_path
```
Comment (VasylYaremkiv) [inline spec/feature/sign_in_spec.rb]: Do not check `path` in view tests.

File: spec/feature/sign_in_spec.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+describe 'Sign in', type: :feature do
+  before(:each) do
+    visit root_path
+  end
+  let(:user) { FactoryBot.create(:user) }
+  it 'correct', js: true do
```
Comment (VasylYaremkiv) [inline spec/feature/sign_in_spec.rb]: Use better name for test. Like `with correct credentials`

File: spec/feature/sign_in_spec.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    expect(page).to have_content 'List of users'
+  end
+  it 'failed', js: true do
+    expect(page).to have_current_path new_user_session_path
+
+    fill_in 'Email', with: user.email
+    click_button 'Log in'
+
+    expect(page).to_not have_current_path root_path
+    expect(page).to_not have_content 'List of users'
```
Comment (VasylYaremkiv) [inline spec/feature/sign_in_spec.rb]: Maybe need to check flash message.

File: spec/feature/sign_up_spec.rb [hunk 8]
```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+describe 'Sign up', type: :feature do
+  let!(:project) { FactoryBot.create(:project) }
+  before(:each) do
+    visit new_user_registration_path
+  end
+  it 'correct', js: true do
```
Comment (VasylYaremkiv) [inline spec/feature/sign_up_spec.rb]: Same here and need to format this file.

File: spec/feature/sign_up_spec.rb [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    {
+      email: 'example@test.com',
+      last_name: 'Test',
+      first_name: 'test',
+      phone: '380972618388',
+      password: 'secret'
+    }
+  end
+
+  def fill_in_sign_up_form(default_options = sign_in_params)
```
Comment (VasylYaremkiv) [inline spec/feature/sign_up_spec.rb]: Move it end of describe and make it private

Comment (VasylYaremkiv) [reply spec/feature/sign_up_spec.rb]: `fill_in_sign_up_form(default_options = {})`

File: spec/feature/sign_up_spec.rb [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    options = sign_in_params.merge(default_options)
+    fill_in 'First name', with: options[:first_name]
+    fill_in 'Last name', with: options[:last_name]
+    fill_in 'Phone', with: options[:phone]
+    fill_in 'Email', with: options[:email]
+    fill_in 'Password', with: options[:password]
+    fill_in 'Password confirmation', with: options[:password]
+    select project.name, from: 'Project'
```
Comment (VasylYaremkiv) [inline spec/feature/sign_up_spec.rb]: `select options[:project], from: 'Project'`

File: spec/feature/sign_up_spec.rb [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  before(:each) do
+    visit new_user_registration_path
+  end
+
+  it 'with correct credentials', js: true do
```
Comment (VasylYaremkiv) [inline spec/feature/sign_up_spec.rb]: `with correct filled form`

File: spec/feature/sign_up_spec.rb [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+    fill_in_sign_up_form
+    click_button 'Sign up'
+
+    expect(page).to have_content 'List of users'
+  end
+
+  it 'with incorrect credentials', js: true do
```
Comment (VasylYaremkiv) [inline spec/feature/sign_up_spec.rb]: `with missed email field`

File: spec/feature/sign_in_spec.rb [hunk 13]
```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+describe 'Sign in', type: :feature do
+  before(:each) do
+    visit new_user_session_path
+  end
+
+  let(:user) { FactoryBot.create(:user) }
```
Comment (VasylYaremkiv) [inline spec/feature/sign_in_spec.rb]: Move let to start of describe block

File: spec/feature/sign_in_spec.rb [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+    click_button 'Log in'
+
+    expect(page).to have_content 'List of users'
+  end
+
+  it 'with incorrect credentials', js: true do
+    fill_in 'Email', with: user.email
+    click_button 'Log in'
+
+    expect(page).to have_content 'Some error message must be here'
```
Comment (VasylYaremkiv) [inline spec/feature/sign_in_spec.rb]: `Some error message must be here` ?

File: spec/feature/sign_up_spec.rb [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+
+  def fill_in_sign_up_form(default_options = {})
+    options = sign_in_params.merge(default_options)
+    fill_in 'First name', with: options[:first_name]
+    fill_in 'Last name', with: options[:last_name]
+    fill_in 'Phone', with: options[:phone]
+    fill_in 'Email', with: options[:email]
+    fill_in 'Password', with: options[:password]
+    fill_in 'Password confirmation', with: options[:password]
+    select project.name, from: 'Project'
```
Comment (VasylYaremkiv) [inline spec/feature/sign_up_spec.rb]: See my comment above:
https://github.com/CyberCraftInc/cyber-dashboard/pull/24#discussion_r245056838

File: spec/feature/select_project_spec.rb [hunk 16]
```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+describe 'Dropdown testing', type: :feature do
```
Comment (VasylYaremkiv) [inline spec/feature/select_project_spec.rb]: `Select project`

File: spec/feature/select_project_spec.rb [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+  let!(:third_user) { FactoryBot.create(:user, project: project_with_user)}
+
+  before(:each) do
+    visit new_user_session_path
+    fill_in 'Email', with: first_user.email
+    fill_in 'Password', with: first_user.password
+    click_button 'Log in'
+  end
+
+  it 'Return all users', js: true do
```
Comment (VasylYaremkiv) [inline spec/feature/select_project_spec.rb]: `returns all users if project is not provided`

File: spec/feature/select_project_spec.rb [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+    visit root_path(id: 0)
+
+    expect(page).to have_content first_user.email
+    expect(page).to have_content second_user.email
+    expect(page).to have_content third_user.email
+  end
+
+  it 'Return 2 users', js: true do
```
Comment (VasylYaremkiv) [inline spec/feature/select_project_spec.rb]: returns all project users if project is provided

File: spec/feature/select_project_spec.rb [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+    visit root_path(id: project_with_2_users.id)
+
+    expect(page).to have_content first_user.email
+    expect(page).to have_content second_user.email
+    expect(page).to_not have_content third_user.email
+  end
+
+  it 'Return 1 user', js: true do
```
Comment (VasylYaremkiv) [inline spec/feature/select_project_spec.rb]: remove this test

File: spec/feature/select_project_spec.rb [hunk 20]
```diff
@@ -0,0 +1,51 @@
...
+    visit root_path(id: project_with_user)
+
+    expect(page).to_not have_content first_user.email
+    expect(page).to_not have_content second_user.email
+    expect(page).to have_content third_user.email
+  end
+
+  it 'Not return users', js: true do
```
Comment (VasylYaremkiv) [inline spec/feature/select_project_spec.rb]: returns empty select if project is provided but project does not have users

**Comment by SetterVet** [comment]
dropdown test working @VasylYaremkiv

**Comment by VasylYaremkiv** [comment]
Looks good. Fix: https://github.com/CyberCraftInc/cyber-dashboard/pull/24/files#r245251144 and squash commits

**Comment by SetterVet** [comment]
What fix? @VasylYaremkiv  in https://github.com/CyberCraftInc/cyber-dashboard/pull/24/files#r245251144

**Comment by SetterVet** [comment]
squashed @VasylYaremkiv @afurm @rubycop

**Pull request merged**